pkg/zfs/zfs_util.go

```diff
@@ -236,13 +249,25 @@ func buildDatasetCreateArgs(vol *apis.ZFSVolume) []string {
 		ZFSVolArg = append(ZFSVolArg, "-o", keyFormat)
 	}
 
-	// set the mount path to none, by default zfs mounts it to the default dataset path
-	ZFSVolArg = append(ZFSVolArg, "-o", "mountpoint=legacy", volume)
-
+	// NOTICE: with mountpoint legacy, the zfs logic which automagically exports
+	// NOTICE: on sharenfs will not work and exporting is left to the operator
+	if vol.Spec.FsType == FSTYPE_NFS {
+		//mountPoint := NFS_ROOT + "/" + vol.Name
+		mountPoint := "/" + vol.Spec.PoolName + "/" + vol.Name
```

Comment:
this mountpoint will be set inside the container. This mountpoint will not be propageted to the host. see the 3rd point here : https://github.com/openebs/zfs-localpv/pull/139#issuecomment-645250165. We need to dedicate one directory from the host and mount that in biderectional mode so that mount within the container can propagate to the host. The mount path can be anything "/var/lib/zfspv/" or anything.

Reply:
Not sure if I read you right: I used the more or less "canonical" ZFS dataset root so that I do not need to care about machines with multiple zpools. Constructing the bidirectional mount correctly then needs to be done when deploying the driver.

You would like a "fixed" path better so that there is no disambiguation regardless whether one or more zpools are present on the machine? Like `"/var/lib/zfspv+  "/" + vol.Spec.PoolName + "/" + vol.Name`?

Reply:
The thing here is the operator will be deployed once and after that user is free to setup the ZFS POOL he wants. So he can setup the  multiple ZPOOL. Like on a node he might have added two kinds of disk - ssd nad nvme. And he might have created two ZFS Pools - zpool-nvme, zpool-ssd. He he will have two storageclasses :-
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: openebs-zfspv
allowVolumeExpansion: true
parameters:
  poolname: "zpool-ssd"
provisioner: zfs.csi.openebs.io
``` 

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: openebs-zfspv
allowVolumeExpansion: true
parameters:
  poolname: "zpool-nvme"
provisioner: zfs.csi.openebs.io
``` 

he can use the storageclasses for his applications. For example the application which need higher IOPS, he will use nvme storageclass and for others applications he cal use ssd storageclass.

How this use case will work here?